docs/upgrade/upgrade-opensource/upgrade-guide-from-5.4-to-6.0/enable-consistent-topology.rst

```diff
@@ -3,4 +3,102 @@ Enable Consistent Topology Updates
+
+  .. code-block:: none
+
+    select value from system.scylla_local where key = 'enabled_features';
+
+  It returns a string which contains a comma-separated list of names of the features that the node sees as enabled. You should wait until it contains ``SUPPORTS_CONSISTENT_TOPOLOGY_CHANGES``.
+
+* Make sure that all nodes are alive for the duration of the upgrade.
+
+.. _enable-raft-topology-6.0-forbidden-operations:
+
+Administrative operations which must not be running during upgrade
+==================================================================
+
+Make sure that administrative operations will not be running while upgrade is in progress.
+In particular, you must abstain from:
+
+* :doc:`Cluster management procedures </operating-scylla/procedures/cluster-management/index>` (adding, replacing, removing, decomissioning nodes etc.).
+* Runing :doc:`nodetool repair </operating-scylla/nodetool-commands/repair>`.
+* Runing :doc:`nodetool checkAndRepairCdcStreams </operating-scylla/nodetool-commands/checkandrepaircdcstreams>`.
+
+Running the procedure
+=====================
+
+.. warning::
+
+  Before proceeding, make sure that all the :ref:`prerequisites <enable-raft-topology-6.0-prerequisites>` are met
+  and no :ref:`forbidden administrative operations <enable-raft-topology-6.0-forbidden-operations>` will run
+  during upgrade. Failing to do so may put the cluster in an inconsistent state.
+
+Starting the upgrade procedure is done by issuing an HTTP request to any of the nodes in the cluster.
+
+For example, you can start the upgrade using the ``curl`` command like this:
+
+.. code-block:: bash
+
+	curl -X POST "http://127.0.0.1:10000/storage_service/raft_topology/upgrade"
+
+Next, wait until all nodes report that upgrade is complete. Checking that a single node finished upgrade can be done in a number of ways:
+
+* With HTTP. It can be done with the same route as the one for triggering upgrade, but by using the ``GET`` method:
+
+  .. code-block:: bash
+
+      curl -X GET "http://127.0.0.1:10000/storage_service/raft_topology/upgrade"
+
+  It will return a JSON string which will be equal to ``done`` after the upgrade is complete on this node.
+
+* By querying the ``system.topology`` table:
+
+  .. code-block:: none
+
+      select upgrade_state from system.topology;
+
+  The ``upgrade_state`` column should be set to ``done`` after the upgrade is complete on this node.
+
+* By looking at the logs. Wait until the following message appears:
+
+  .. code-block:: none
+
+      raft_topology - upgrade to raft topology has finished
+
+After the upgrade is complete on all nodes, wait for the duration of ``ring_delay_ms`` before issuing any topology changes in order to avoid data loss from writes that were started before the upgrade.
+
+What if upgrade gets stuck?
+===========================
+
+If the process gets stuck at some point, perform :ref:`the Raft recovery procedure <recovery-procedure>`.
```

Comment:
Before referring to the recovery procedure, we should ask the admin to perform the standard checks, i.e. are all nodes UP? restart them if you can etc.